apps/ui/components/common/Navigation.tsx

```diff
@@ -109,29 +120,51 @@ function constructName(firstName?: string, lastName?: string) {
 }
 
 function NavigationMenu({ user }: { user: CurrentUserQuery["currentUser"] }) {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const { pathname } = useLocation();
+  const router = useRouter();
+
+  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    e.preventDefault();
+    router.push(e.currentTarget.href);
+  };
+
   return (
     <Header.NavigationMenu>
       {menuItems.map((item) => {
         if (item.requireLogin && !user) {
           return null;
         }
+        const localisationString =
+          i18n.language === "fi" ? "" : getLocalizationLang(i18n.language);
         const isActive = () => {
           if (!pathname) return false;
-          let active = false;
-          item.href.forEach((href) => {
-            if (pathname.startsWith(href)) {
-              active = true;
+          for (let route of item.routes) {
+            const localizedPath =
+              localisationString !== ""
+                ? `/${localisationString}${route}`
+                : route;
+            if (
+              item.exact
+                ? pathname === localizedPath
+                : pathname.startsWith(localizedPath)
+            ) {
+              return true;
             }
-          });
-          return active;
+          }
+          return false;
         };
```

Comment:
this should be in the component root (or a free function), don't define functions inside loops